.changelog/3342.bugfix.md

```diff
@@ -0,0 +1 @@
+go/executor/node: ensure correct state before publishing a message
```

Comment:
Could be changed to trivial as it fixes something that was not released yet.

Reply:
oh right, yeah agreed, otherwise it's probably confusing